### DISCUSSION TITLE
Is there a running list of which U-Net parameters work well?

### DISCUSSION THREAD

**Discussion by oconnoob** [lucidrains/imagen-pytorch#177] [Q&A]
Hey everyone! I'm currently working on training an Imagen model, and I'm wondering if there are any guidelines / a compilation of parameters that work well?

There's a lot of flexibility in how the U-Nets are parameterized, so guidance on that would be super helpful :)

**Comment by lucidrains** [comment]
@oconnoob why not just attempt what they had in the paper? https://github.com/lucidrains/imagen-pytorch/blob/main/imagen_pytorch/imagen_pytorch.py#L1644

**Reply by oconnoob** [reply]
@lucidrains I stripped down this implementation into a [minimal one](https://github.com/AssemblyAI-Examples/MinImagen) for educational purposes, and in the process lost some training efficiency, so I've found the original paper parameters to be a bit slow to train.

I might just have to bite the bullet and do that however! Thanks for the suggestion
